File: internal/utils/email.go
package utils

import (
	"fmt"
	"net/smtp"

	"wavy-backend/internal/config"
)

func SendOTPEmail(cfg *config.Config, to string, code string) error {
	subject := "Kode OTP Wavy Anda"
	body := fmt.Sprintf("Kode OTP Anda: %s\n\nBerlaku selama 5 menit. Jangan bagikan kode ini ke siapapun.", code)

	msg := []byte("To: " + to + "\r\n" +
		"Subject: " + subject + "\r\n" +
		"\r\n" + body + "\r\n")

	auth := smtp.PlainAuth("", cfg.SMTPUser, cfg.SMTPPass, cfg.SMTPHost)
	addr := cfg.SMTPHost + ":" + cfg.SMTPPort

	return smtp.SendMail(addr, auth, cfg.SMTPFrom, []string{to}, msg)
}

File: src/utils/email.ts
import nodemailer from "nodemailer";

const transport = nodemailer.createTransport({
  host: process.env.SMTP_HOST,
  port: Number(process.env.SMTP_PORT),
  secure: false,
  auth: {
    user: process.env.SMTP_USER,
    pass: process.env.SMTP_PASS,
  },
});

export async function sendOtpEmail(to: string, code: string) {
  await transport.sendMail({
    from: process.env.SMTP_FROM,
    to,
    subject: "Kode OTP Wavy Anda",
    text: `Kode OTP Anda: ${code}\nKode berlaku selama 5 menit. Jangan bagikan kode ini kepada siapa pun.`,
  });
}